src/components/DemoBanner/DemoBanner.tsx

```diff
@@ -0,0 +1,135 @@
+import ExternalLink from "@saleor/components/ExternalLink";
+import { makeStyles } from "@saleor/macaw-ui";
+import React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { emphasisedTextBlue, greyDark, textColor } from "./constants";
+export const mediumScreen = 720;
+
+const useStyles = makeStyles(
+  theme => ({
+    wrapper: {
+      width: "100%",
+      backgroundColor: "inherit",
+      padding: `0 13px`,
+      fontSize: theme.spacing(1.5)
+    },
+    borderedWrapper: {
+      display: "flex",
+      flexDirection: "row",
+      justifyContent: "space-between",
+      alignItems: "center",
+      maxHeight: theme.spacing(7.5),
+      flex: "0 0 auto",
+      padding: theme.spacing(3, 0),
+      backgroundImage: `linear-gradient(to right,  ${greyDark} 13%, rgba(255, 255, 255, 0) 0%)`,
+      backgroundPosition: "bottom",
+      backgroundSize: theme.spacing(1.5, 0.125),
+      backgroundRepeat: "repeat-x"
+    },
+    logoWrapper: {
+      lineHeight: 0
+    },
+    linkList: {
+      display: "flex",
+      alignItems: "center"
+    },
+    link: {
+      display: "flex",
+      padding: theme.spacing(2),
+      color: textColor
+    },
+    textEmphasis: {
+      color: emphasisedTextBlue,
+      textTransform: "uppercase",
+      fontWeight: 600,
+      paddingLeft: "5px"
+    },
+    divider: {
+      borderRight: `1px solid  ${greyDark}`,
+      height: theme.spacing(2)
+    },
+    mediaMedium: {
+      display: "flex",
+      [theme.breakpoints.up(mediumScreen)]: {
+        display: "none"
+      }
+    },
+    mediaLarge: {
+      display: "flex",
+      alignItems: "center",
+      [theme.breakpoints.down(mediumScreen)]: {
+        display: "none"
+      }
+    }
+  }),
+  {
+    name: "DemoBanner"
+  }
+);
+
+export const DemoBanner: React.FC = () => {
+  const classes = useStyles();
+
+  return (
+    <div className={classes.wrapper}>
+      <div className={classes.borderedWrapper}>
+        <div />
+        <div className={classes.linkList}>
+          <div className={classes.mediaMedium}>
+            <ExternalLink
+              className={classes.link}
+              href={"https://demo.saleor.io/graphql/"}
+            >
+              <div className={classes.textEmphasis}>
+                <FormattedMessage defaultMessage="API" />
+              </div>
+            </ExternalLink>
+            <ExternalLink
+              className={classes.link}
+              target="_blank"
+              href="https://demo.saleor.io/dashboard/"
+            >
+              <div className={classes.textEmphasis}>
+                <FormattedMessage defaultMessage="Dashboard" />
+              </div>
+            </ExternalLink>
+          </div>
+
+          <div className={classes.mediaLarge}>
```

Comment:
The task was to copy the code from different app, and amend when necessary – like conversion from Styled Components to Material UI. Not going line-by-line to seek and implement improvements. 
If other major changes are implemented here, it would be good to amend it in the other app as well.
I would suggest to keep changes only to a necessary level here, and then create a ticket to introduce improvements in both versions in future. 
This comment applies also to suggestions related to links.

Reply:
I believe that writing this answer took you longer than applying suggested amendments. You're not copying 3000 LOCs, it's a small component that could be written in 5 minutes.